core/src/main/java/org/mapfish/print/output/MapExportOutputFormat.java

```diff
@@ -0,0 +1,197 @@
+    private String contentType;
+        
+    public final void setContentType(final String contentType) {
+        this.contentType = contentType;
+    }
+
+    @Override
+    public final String getContentType() {
+        return this.contentType;
+    }
+
+    public final void setFileSuffix(final String fileSuffix) {
+        this.fileSuffix = fileSuffix;
+    }
+
+    @Override
+    public final String getFileSuffix() {
+        return this.fileSuffix;
+    }
+
+    /**
+     * One of {@link java.awt.image.BufferedImage} TYPE_ values.
+     *
+     * @param imageType the buffered image type to create.
+     */
+    public final void setImageType(final int imageType) {
+        this.imageType = imageType;
+    }
+    
+    private String getMapSubReportVariable(final Template template) {
+        for (Processor<?, ?> processor : template.getProcessors()) {
+            if (processor instanceof CreateMapProcessor) {
+                String mapSubReport = ((CreateMapProcessor) processor).getOutputMapperBiMap().get(MAP_SUBREPORT);
+                if (mapSubReport == null) {
+                    return MAP_SUBREPORT;
+                } else {
+                    return mapSubReport;
+                }
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public final Print getJasperPrint(final PJsonObject requestData, final Configuration config,
+            final File configDir, final File taskDirectory)
+                    throws JRException, SQLException, ExecutionException, JSONException {
+        final String templateName = requestData.getString(Constants.JSON_LAYOUT_KEY);
+
+        final Template template = config.getTemplate(templateName);
+                
+        final Values values = new Values(requestData, template, this.parser, taskDirectory, this.httpRequestFactory, null);
+                
+        final ForkJoinTask<Values> taskFuture = this.forkJoinPool.submit(template.getProcessorGraph().createTask(values));
+
+        try {
+            taskFuture.get();
+        } catch (InterruptedException exc) {
+            // if cancel() is called on the current thread, this exception will be thrown.
+            // in this case, also properly cancel the task future.
+            taskFuture.cancel(true);
+            Thread.currentThread().interrupt();
+            throw new CancellationException();
+        }
+        
+        String mapSubReport = values.getString(getMapSubReportVariable(template));        
+        if (mapSubReport == null) {
+            throw new IllegalArgumentException("Missing CreateMap process in template  " + templateName + ".");
+        }
+        
+        //convert URI to file path
+        try {
+            mapSubReport = new File(new URI(mapSubReport)).getAbsolutePath();
+        } catch (URISyntaxException e) {
+            throw new IllegalStateException(e); //can't really happen
+        }
+        
+        JasperFillManager fillManager = getJasperFillManager(config);
```

Comment:
Yes, it's true that the map-subreport makes it easy to create a graphic file with the report. But I think it would be a good improvement to directly merge the layer graphics without involving the JasperReports engine.

Any opinion, @sbrunner or @pvalsecc?